### DISCUSSION TITLE
How to use PostCSS in Next.js?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#65625] [Help]
### Summary

I get build errors from the Vercel site ... but not in the console when I do `npm run build`

I am trying to get rid of this error from the console:
`Reduce unused CSS`
![image](https://github.com/vercel/next.js/assets/93617433/c3db7a3f-0790-4274-93b6-fb12cdf0a27e)

which sends me to this site:
https://purgecss.com/guides/next.html?utm_source=lighthouse&utm_medium=devtools

Tells me to:
1. create a  postcss.config.js  at the root
2. npm install postcss-flexbugs-fixes postcss-preset-env.
3.  Add the code below to the config file:

```
module.exports = {
  "plugins": [
    "postcss-flexbugs-fixes",
    [
      "postcss-preset-env",
      {
        "autoprefixer": {
          "flexbox": "no-2009"
        },
        "stage": 3,
        "features": {
          "custom-properties": false
        }
      }
    ],
    [
      '@fullhuman/postcss-purgecss',
      {
        content: [
            './pages/**/*.{js,jsx,ts,tsx}',
            './components/**/*.{js,jsx,ts,tsx}'
        ],
        defaultExtractor: content => content.match(/[\w-/:]+(?<!:)/g) || [],
        safelist: ["html", "body"]
      }
    ],
  ]
}

```

^^^  I do all of that above and ...

npm run build
npm run start
..  and the error still there... also... it won't let me actually upload it to vercel to try to test a live site.

Doing the coverage from the developers console... I think it comes from all the CSS from bootstrap... but I don't know which CSS I am using, and which CSS is extra, so I need to fix it.... but do I go about doing it?

![image](https://github.com/vercel/next.js/assets/93617433/ca45c9d7-cdc4-4039-be63-258e068ebe58)

**Comment by xiaolannuoyi** [comment]
你好，你的问题解决了么？求解
